### DISCUSSION TITLE
I've made updated VSCode Lua Snippets for the TIC-80 API

### DISCUSSION THREAD

**Discussion by James-Fraser-Jones** [nesbox/TIC-80#1713] [Show and tell]
I found the existing snippets file pointed to by the [website](https://gist.github.com/Viza74/40a180155049dd26af378f51a92b6033) to be lacking. So I wrote a [new snippets file](https://gist.github.com/smudgecat123/0bf42bd35131528855ff055348d7f28f) for all the API functions currently listed in the wiki. I've also included option prompts to indicate default values and value ranges for certain parameters.

If this is useful to people, perhaps it could be added to the website?

Also, if there is interest, I could attempt to repackage these snippets as a fully fledged VSCode extension to make it slightly easier for users to install.

**Comment by joshgoebel** [comment]
This sounds like a great idea, but Wren is my language of choice... I wonder if we had a TIC-80 extension if we should try to consolidate all TIC-80 support in a single extension (that might ship with multiple languages, etc)... ie. if a full fledge extension is created should it be "TIC-80 support" or "TIC-80 Lua support" only?

**Reply by James-Fraser-Jones** [reply]
That's a good point. It probably would make more sense to have support for all languages within one extension. In that case, I would probably consider trying to write a more comprehensive TIC-80 extension including support for VSCode's "[programmatic language features](https://code.visualstudio.com/api/language-extensions/overview#programmatic-language-features)".
